packages/ckit/src/providers/mercury/MercuryProvider.ts

```diff
@@ -24,21 +24,43 @@ function toCell(resolved: ResolvedOutpoint): Cell {
   };
 }
 
+interface BatchRequest {
+  method: string;
+  params: unknown;
+}
+
 export class MercuryProvider extends AbstractProvider {
   readonly mercury: MercuryClient;
   readonly rpc: RPC;
+  readonly rpcUrl: string;
 
-  constructor(
-    mercuryRpc: MercuryClient | string = 'http://127.0.0.1:8116',
-    ckbRpc: RPC | string = 'http://127.0.0.1:8114',
-  ) {
+  constructor(mercuryRpc = 'http://127.0.0.1:8116', ckbRpc = 'http://127.0.0.1:8114') {
     super();
 
-    if (mercuryRpc instanceof MercuryClient) this.mercury = mercuryRpc;
-    else this.mercury = new MercuryClient(mercuryRpc);
+    this.mercury = new MercuryClient(mercuryRpc);
+    this.rpc = new RPC(ckbRpc);
+    this.rpcUrl = ckbRpc;
+  }
+
+  async batchRequestCkb<T>(request: BatchRequest[]): Promise<T[]> {
+    const batch = [];
+    for (let i = 0; i < request.length; i++) {
+      batch.push({
+        id: i,
+        jsonrpc: '2.0',
+        method: request[i]!.method,
+        params: [request[i]!.params],
+      });
+    }
+    const res = await fetch(this.rpcUrl, {
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(batch),
+    });
 
-    if (ckbRpc instanceof RPC) this.rpc = ckbRpc;
-    else this.rpc = new RPC(ckbRpc);
+    return await res.json();
```

Comment:
maybe we should unwrap the JSON RPC result here to keep the interface easy to use

Reply:
Not sure if my understanding is right, if we don't pass id back, why bother making it incremental?

Reply:
if id checking is needed, we can just impl it in batchRequest method